src/test/kotlin/com/music/awesomemusic/controllers/SignInTest.kt

```diff
@@ -0,0 +1,127 @@
+        accountRepository.deleteAll()
+    }
+
+
+    @Test
+    @Throws(Exception::class)
+    fun shouldSignInWithUsernameExpected200() {
+
+        accountService.createAccount(AccountSignUpForm("test1", "1234", "emailTest1", false))
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_JSON
+        val request = HttpEntity<String>("{\"login\": \"test1\", \"password\": \"1234\"}", headers)
+        val responseEntity = restTemplate.postForEntity("http://localhost:$port/api/user/sign-in", request, String::class.java)
+
+        assertEquals(responseEntity.statusCode, HttpStatus.OK)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun shouldSignInWithEmailExpected200() {
+
+        accountService.createAccount(AccountSignUpForm("test2", "1234", "emailTest2", false))
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_JSON
+        val request = HttpEntity<String>("{\"login\": \"emailTest2\", \"password\": \"1234\"}", headers)
+        val responseEntity = restTemplate.postForEntity("http://localhost:$port/api/user/sign-in", request, String::class.java)
+
+        assertEquals(responseEntity.statusCode, HttpStatus.OK)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun shouldSignInWithEmailIfNoExistsExpected403() {
+
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_JSON
+        val request = HttpEntity<String>("{\"login\": \"emailTest3@gmail.com\", \"password\": \"1234\"}", headers)
+        val responseEntity = restTemplate.postForEntity("http://localhost:$port/api/user/sign-in", request, String::class.java)
+
+        assertEquals(responseEntity.statusCode, HttpStatus.FORBIDDEN)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun shouldSignInWithUsernameIfNoExistsExpected403() {
+
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_JSON
+        val request = HttpEntity<String>("{\"login\": \"testName\", \"password\": \"1234\"}", headers)
+        val responseEntity = restTemplate.postForEntity("http://localhost:$port/api/user/sign-in", request, String::class.java)
+
+        assertEquals(responseEntity.statusCode, HttpStatus.FORBIDDEN)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun shouldSignInReturnTokenExpected200() {
+
+        val account = accountService.createAccount(AccountSignUpForm("testName", "1234", "emailTest10", false))
+        val authorities = arrayListOf<String>()
+
+        val token = tokenProvider.createToken(account.username, authorities)
+
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_JSON
+
+        val request = HttpEntity<String>("{\"login\": \"testName\", \"password\": \"1234\"}", headers)
+        val responseEntity = restTemplate.postForEntity("http://localhost:$port/api/user/sign-in", request, String::class.java)
+
+        assertEquals(tokenProvider.validateToken(token), true)
```

Comment:
You didn't check token from response. You've check private method "tokenProvider.createToken", not public "/sign-in"